### DISCUSSION TITLE
AOSP 11.0 v308

### DISCUSSION THREAD

**Discussion by phhusson** [phhusson/treble_experimentations#1850] [General]
- Fix touchscreen on some OPPO devices
- Fix boot and screen orientation on BQ Aquaris M10 (thanks @R0rt1z2 )
- Fix Brightness, FP and Headphone jack issue of Realme Narzo 20 and Realme X7 Pro 5G (thanks @sarthakroy2002)

Treble_presets (those changes are automatically updated without upgrading GSI):
- LG Velvet: Better match devices (Thanks @dom133 )
- LG Velvet: Fix wired and bluetooth headset (Thanks @dom133 )

<hr /><em>This discussion was created from the release <a href='https://github.com/phhusson/treble_experimentations/releases/tag/v308'>AOSP 11.0 v308</a>.</em>

**Comment by realKarthikNair** [comment]
Thank you

**Comment by eltaba16** [comment]
Do you add nav bar inverse? Thanks!!

**Reply by realKarthikNair** [reply]
Nav bar inverse isnt part of AOSP Source code, so NO

**Comment by sangare050** [comment]
Bro on my infinix I can use the earphones but the sound is not mute when I put the earphone on: that is to say that when the earphone is plugged in while listening to music the phone as the earphone plays the music

**Comment by filmwashowa** [comment]
why i can root with magisk on my devices when i using your rom, my device is vsmart active 3 with mediatek helio p60 chip

**Reply by realKarthikNair** [reply]
u can or u cant?

**Comment by Marcos8760** [comment]
I don't wanna be pushy at all, but are you working on an Android 12 GSI (beta)(to be released in the coming months)? Or are you waiting for a full release, or a later beta release? (Google I/O was really cool this year haha)

Thanks!

- L

**Comment by phhusson** [comment]
There is no Android 12 source code. Android 12 basically doesn't exist for me at the moment

24 mai 2021 17:28:04 Marcos8760 ***@***.***>:

> I don't wanna be pushy at all, but are you working on an Android 12 GSI (beta)(to be released in the coming months)? Or are you waiting for a full release, or a later beta release? (Google I/O was really cool this year haha)
> 
> Thanks!
> 
* > L

**Reply by Marcos8760** [reply]
Alright i gotchu. 

Thanks!

- L

**Comment by BlakeU666** [comment]
Uhm, the phone is spamming "This device doesn't have the Play Certificate".

**Comment by phhusson** [comment]
Register your device on https://www.google.com/android/uncertified/

To get the ID, you can either:
- adb root, then adb shell, press enter, and at the new prompt, inside adb shell, do sqlite3 /data/data/com.google.android.gsf/databases/gservices.db    'select * from main where name = "android_id";'
- https://play.google.com/store/apps/details?id=com.evozi.deviceid => Google Service Framework (GSF) ID
- for multiuser see https://github.com/phhusson/treble_experimentations/issues/1794#issuecomment-816632811

24 mai 2021 20:27:02 BlakeU666 ***@***.***>:

**Comment by whocker** [comment]
There is an issue with Contacts Storage (ContactsProvider.apk). For some reason, when I try to save a contact to my device, it does not work. Google Contacts does not allow the saving to phone storage, likely because ContactsProvider.apk has a problem

**Comment by kutlay1653** [comment]
Any change for getting arm64 go?
Thanks

**Reply by realKarthikNair** [reply]
Any change?

**Reply by realKarthikNair** [reply]
what did you mean by "change"?

**Reply by kutlay1653** [reply]
> Any change?

Well I mean chance not change sorry. Keyboard thx.




I believe we need aosp 11 go arm64 gsi that's all

**Comment by aksbra** [comment]
Sir how can i root your rom 
And one more thing i didn't tried your rom yet but i just wanna know that i install your rom by tommorow and yesterday i tried pixel experience rom but it says partition space is not enough. Will this work for my device 
System partition=1301328
And one more thing why is there only go apps?

**Reply by realKarthikNair** [reply]
```
Warning ! Dont try this if you don't know anything about android

Guide to get magisk root working on @phhusson android 11 gsi

1.install termux
2.enter su
3.enter echo $PATH
4.copy the output path and paste inside init.environ.rc inside #root directory in the format "export PATH <that path>"
(Use a files manager like Mixplorer or anything which can rw root)
5.if there's no sbin folder inside root , create one
6.flash magisk 21.x and reboot
7. you'll see magisk 21.x is installed
8. update magisk manager to 22.0 from 21.x manager , then using newer manager , update magisk to 22.0 using direct method

Bingo!
```

**Reply by filmwashowa** [reply]
can you make a video tutorial, please????

**Reply by Meliox** [reply]
Thank you for the instruction. Should be added to the Wiki :)

**Reply by Hausemaster8281** [reply]
> ```
> Warning ! Dont try this if you don't know anything about android
> 
> Guide to get magisk root working on @phhusson android 11 gsi
> 
> 1.install termux
> 2.enter su
> 3.enter echo $PATH
> 4.copy the output path and paste inside init.environ.rc inside #root directory in the format "export PATH <that path>"
> (Use a files manager like Mixplorer or anything which can rw root)
> 5.if there's no sbin folder inside root , create one
> 6.flash magisk 21.x and reboot
> 7. you'll see magisk 21.x is installed
> 8. update magisk manager to 22.0 from 21.x manager , then using newer manager , update magisk to 22.0 using direct method
> 
> Bingo!
> ```

The issue with this method is that, the file init.environ.rc is reverted to the original state after a reboot, to put it simply it doesn't persist across reboot, and trying this method only ends up in the device not booting unless the stock boot image is restored, so, please share any ideas you have about getting magisk working...

**Reply by realKarthikNair** [reply]
> The issue with this method is that, the file init.environ.rc is reverted to the original state after a reboot, to put it simply it doesn't persist across reboot, and trying this method only ends up in the device not booting unless the stock boot image is restored, so, please share any ideas you have about getting magisk working...

wait a second, after editing init.environ.rc, your device failed to boot?
afaik, this change persists even after rebooting (if your device has booted successfully after editing the file)

**Reply by realKarthikNair** [reply]
> Thank you for the instruction. Should be added to the Wiki :)

this workaround is known since over an year now, but they don't add it to the wiki . i Dont really know why

**Reply by phhusson** [reply]
@realKarthikNair you know you're part of "they" right?

**Reply by realKarthikNair** [reply]
> @realKarthikNair you know you're part of "they" right?

Oh! Sorry, I'll add it.

**Comment by aksbra** [comment]
Bingo but will su command work without root?

**Reply by realKarthikNair** [reply]
You already have a su binary bundled in phhusson gsi

**Comment by aksbra** [comment]
Okay i dont want root right now
I am facing another problem
I factory reset my device then went to fastboot then erased system and then
i flashed your androind 11 gsi but when i reboot it stuck at logo. Then i
thought it can be problem of my boot.img that i flashed when rooting so i
flashed stock boot.img and boom my device booted but when it restarts i
mean start it stuck at lock screen i cant do anything
What is problem there do i have to flash any other patch or i have to
install google aaps?
What i have to do?

**Reply by realKarthikNair** [reply]
please specify these factors:

1. Which varient did you flash?
2. Your device and model number 
3. by "lock screen" , did you mean setup wizard ?
4. Did the bootanimation have "android" written on it?

**Reply by aksbra** [reply]
Ok so i used latest android 11 version
My device - lenovo tb-7104f tablet running on android 8.1 oreo meditek
mt8167d processor

The problem is when i flash your img file through fastboot (i dont have any
recovery) i easily boot in img it shows android logo and then it starts but
when it stars it stuck at lock screen i mean there is not any problem in
boot it boots normally but after booting i cant use it it just showing me
lock screen again and again

Can i try android 10 version first?

Is google apps must to run it?
…

**Reply by CoelusCrux** [reply]
No google apps are not necessary. Try android 10 or 9.

**Comment by aksbra** [comment]
Sir i did many thing but i still cant fix it i am stuck at lock screen 


https://user-images.githubusercontent.com/84490914/120055808-03e51f80-c056-11eb-9bbf-19824ea5821c.mp4

**Reply by CoelusCrux** [reply]
Try android 10 or less.

**Reply by realKarthikNair** [reply]
try launching apps from adb

**Reply by aks-glitch** [reply]
How is it possible?

**Comment by MacariaSonata** [comment]
ummm in the previous version, theres a problem with microphone, i wonder if its been fixxed in this version?

problem : sound input was too low when im in call/ either called by other people or it wont input any sound at all when i talk, i must clear dalvik cache to make it work again, but still voice is too low

Phone : Lenovo K5 Play
Installed version : Arm64 A Only Vanilla
Installed Gapps : NikGapps

**Reply by realKarthikNair** [reply]
First of all, its recommended to open an issue at the project's page, not here,

regarding low volume, try disabling a2dp overload in phh settings>misc

And installing the gapps included gsi would be a better option than flashing gapps over vanilla

**Reply by gombalr99** [reply]
Can you teach me to flash this rom?, I have the same phone as you, but the phone won't boot

**Comment by CoelusCrux** [comment]
Not able to remove supersu from settings securise option. Any way to remove it?
Model- honor 7X 
GSi - arm64ab gapps v308

**Reply by realKarthikNair** [reply]
what exactly happens when you try to to use the securize option?

**Reply by CoelusCrux** [reply]
Pop-up for confirmation, after accepting it phone restarts

**Comment by Meliox** [comment]
Anyone able to get Widewine L1 working to be able to play Netflix in HD?

**Reply by realKarthikNair** [reply]
some OEMs downgrade WIdevine to L3 once Bootloader is unlocked, so blame your vendor.

**Reply by Hausemaster8281** [reply]
Try looking for a vendor image without the restrictions, this depends on your device...

**Comment by Hausemaster8281** [comment]
Did Google release the source code for Android 11 go edition?
(The Nokia 1 plus got the Android 11 go update yesterday BTW)

**Reply by realKarthikNair** [reply]
no
